### ISSUE TITLE
Confused on usage

### ISSUE THREAD

**Issue by cpebble** [xxxserxxx/gotop-nvidia#4]
I gotop as a dashboard on my secondary monitor, and i saw that the nvidia extension has been released. I felt adventurous and tried to build it myself and it seems like i succeeded, _except_: 
The only output i get from the nvidia extension is temperature. CPU Usage still only lists 0-5 and Memory usage only shows Main and Swap. See attached image for the output of my current gotop, using default layouts.  

Running the parsed nvidia-smi command shows that all output is as expected:  

    $ nvidia-smi --query-gpu=name,index,temperature.gpu,utilization.gpu,memory.total,memory.used
    --format=csv,noheader,nounits
    GeForce GTX 1080 Ti, 0, 31, 9, 11175, 206

![Gotop look](https://user-images.githubusercontent.com/9165540/102374320-7d65e580-3fc1-11eb-80be-560d9814d68b.png)

**Comment by xxxserxxx** [comment]
This looks like what @talentlessguy reported in #1 .  I don't have an NVidia card to test, so it's a lot of trial-and-error.  I'll revisit this with some code to check the results from the NVidia code.

**Comment by cpebble** [comment]
Hello @xxxserxxx, i have done some bug hunting(throwing cpu values at the metaphorical wall) and managed to get the values to display correctly. Now i'll test this out as a daily driver for a few days but it works for me(TM)

The fork that works for me is available at [https://github.com/cpebble/gotop-nvidia](https://github.com/cpebble/gotop-nvidia)

_However_ there is a catch. I have no idea how this works. I found out that the values display properly when i initialize
them in the `startup` function. So my fix specifically is adding these  [two lines](https://github.com/cpebble/gotop-nvidia/blob/af47c44ff2ce79491cd1e601a6944ee9aea32f6a/nvidia.go#L65) 
This adds the issue that we need to know the GPU Name beforehand.

I also don't know the go language very well, so i have no way to know why the `map` behaves like this or why this bug may occur. I would like some feedback on what's going on

**Comment by xxxserxxx** [comment]
@cpebble This is great; thanks for the effort. Do you want to submit a pull request? It doesn't matter if it's not pretty -- right now, I'm programming blind on nvidia, and it's easier to review and diff something that I know works.

**Comment by xxxserxxx** [comment]
@cpebble fixed by 68eea8c2858b8a74804bd4bbfae0593e745aa59d

The issue is that the widgets need to know what the labels are when they're set up; in the case of the nvidia plugin, this wasn't happening.

The plugins are confusing; I acknowledge this. The reason they are the way they are is outlined in the gotop wiki. In any case, gotop requests data for each device -- CPUs, memory, temperature... and since nvidia's data comes from a command, it would mean that on every refresh the nvidia-smi command would be forked and parsed three times.  To make this more efficient, the plugin forks off a process to exec the command periodically, parse it once, and cache the output. Then when gotop asks for the data, the plugin just returns the cached values. It makes things more efficient, but complicates the code. What was happening was that the initial setup call was returning before there was any cached data, so gotop thought there were no stats (except temperature, which doesn't need to know the labels ahead of time).

I don't know how you were building this, but here's what I recommend:

```
mkdir gotop-nvidia
cd gotop-nvidia
go get github.com/xxxserxxx/gotop-builder
gotop-builder -r v4.1.0 github.com/xxxserxxx/gotop-nvidia
echo 'replace github.com/jdkeke142/lingo-toml => github.com/xxxserxxx/lingo v0.0.0-20210125170606-5858cf2e409d' >> go.mod
go build -o gotop .
```

**Comment by xxxserxxx** [comment]
I pushed it out as a release; it's the 4.1.0 tag.  Please do let me know if it works for you.
